### ISSUE TITLE
VM names are not URL encoding special characters

### ISSUE THREAD

**Issue by Epaphus** [synackray/vcenter-netbox-sync#90]
### Environment
* Python version:  3.6.8
* NetBox version: 2.7.10
* vCenter version: 6.5
* vCenter NetBox Sync version:  Github master branch

### Steps to Reproduce
1. Have a VM with name which has character which are normally used in URL query strings such as '&'
2. Run a sync

### Expected Behavior
I would expect all vCenter VMs to sync with netbox

### Observed Behavior
When sync gets to a VM with "&" in the name the sync fails.

```
2020-04-03 16:58:08,422 [INFO] NetBox successfully created virtual_interfaces object.
2020-04-03 16:58:08,422 [DEBUG] Sending GET to 'https://netbox/api/virtualization/interfaces/?virtual_machine=abcd&efghi&name=vNIC0' with data 'None'.
2020-04-03 16:58:08,438 [DEBUG] Received HTTP Status 400.
2020-04-03 16:58:08,438 [CRITICAL] Well this in unexpected. Please report this. GET request received 400 status with body 'None' and response '{'virtual_machine': ['Select a valid choice. You is not one of the available choices.']}'.
```

The VM name is set by the customer via vCloud Director so isn't something we have control of.

If I encode the ampersand correctly and visit the URL in in browser it works fine.

**Comment by inetman28** [comment]
I have same issue when vm name have a space.

**Comment by synackray** [comment]
Hi, All!

I will do some testing with this and report back. Thanks for the report!

**Comment by alexgoesgit** [comment]
Quick'n dirty hack, that worked for me:

Edit run.py:
```
16a17
> import urllib
947a949
>
1058c1060
<                 vc_data["virtual_machine"]["name"], query_key,
---
>                 urllib.parse.quote(vc_data["virtual_machine"]["name"]), query_key,
```

**Comment by Epaphus** [comment]
Almost the same as what I came up with the other day with a bit of Googling. I haven't done much with Python so wasn't sure if was the best way.

```
import urllib

<                 vc_data["virtual_machine"]["name"], query_key,
---
>                 urllib.parse.quote_plus(vc_data["virtual_machine"]["name"]), query_key,

```

**Comment by synackray** [comment]
Apologies for the delay in getting this sorted. The current state of affairs has made maintaining projects challenging. This should be patched in af762e111481387311301ffdb61d03fcd25ea7ab. It has passed initial testing and will be merged into `develop` shortly before a new `master` release.
